pkg/controller/azdrivernode.go

```diff
@@ -88,25 +91,40 @@ func (r *ReconcileAzDriverNode) Reconcile(ctx context.Context, request reconcile
 
 // run an update on existing azdrivernode objects to store them under new version if necessary
 func (r *ReconcileAzDriverNode) Recover(ctx context.Context) error {
+	var errs []error
 	var err error
 	ctx, w := workflow.New(ctx)
 	defer func() { w.Finish(err) }()
 
 	var nodes *azdiskv1beta2.AzDriverNodeList
 	if nodes, err = r.azClient.DiskV1beta2().AzDriverNodes(r.objectNamespace).List(ctx, metav1.ListOptions{}); err != nil {
-		if errors.IsNotFound(err) {
-			return nil
-		}
 		return err
 	}
 
 	for _, node := range nodes.Items {
-		updated := node.DeepCopy()
-		updated.Annotations = azureutils.AddToMap(updated.Annotations, consts.RecoverAnnotation, "azDriverNode")
-		if _, err = r.azClient.DiskV1beta2().AzDriverNodes(r.objectNamespace).Update(ctx, updated, metav1.UpdateOptions{}); err != nil {
-			return err
+		// if the corresponding node has been deleted, delete the azdriverode object
+		_, err := r.kubeClient.CoreV1().Nodes().Get(ctx, node.Spec.NodeName, metav1.GetOptions{})
+		if err != nil && errors.IsNotFound(err) {
```

Comment:
If `err != nil` but isn't a "not found" error, we'll end up in the else clause and update the `AzDriverNode` with the recover annotation. Is this correct? We won't really know the state of the node. I think we should just log the error and move on to the next node.

Reply:
Okay, I thought it'd be saver to still recover the CRI if we can't confirm the node doesn't exist. I'll change it to log the error and keep the CRI without recovery.